forward/request_sender.go

```diff
@@ -208,13 +236,82 @@ func (s *requestSender) MakeCall(ctx context.Context, res *[]byte, fwdError *err
 			return
 		}
 
-		*res = arg3
 		done <- true
 	}()
 
 	return done
 }
 
+func forwardArgs(call *tchannel.OutboundCall, arg2, arg3 []byte, res *[]byte) error {
+	if err := writeArg(call.Arg2Writer, arg2); err != nil {
+		return err
+	}
+	if err := writeArg(call.Arg3Writer, arg3); err != nil {
+		return err
+	}
+
+	resp := call.Response()
+	if err := discardArg(resp.Arg2Reader); err != nil {
+		return err
+	}
+
+	buf := bufPool.Get()
+	defer bufPool.Put(buf)
+	if err := bufReadArg(buf, resp.Arg3Reader); err != nil {
+		return err
+	}
+
+	if res == nil {
+		*res = []byte{}
+	}
+	*res = append(*res, buf.Bytes()...)
+	return nil
+}
+
+func writeArg(writable func() (tchannel.ArgWriter, error), bs []byte) error {
+	aw, err := writable()
+	if err != nil {
+		return err
+	}
+	_, err = aw.Write(bs)
+	if err == nil {
+		err = aw.Close()
+	}
+	return err
+}
+
+func discardArg(readable func() (tchannel.ArgReader, error)) error {
+	var null [32 * 1024]byte
+	ar, err := readable()
+	if err != nil {
+		return err
+	}
+	for {
```

Comment:
can we use `ioutil.Discard.ReadFrom(..)` here?

Reply:
TIL; will do